Refactor of the code

stebranchi opened the pull request:
I think this could be a slightly better code

TKasekamp commented on jobs/serializers.py:
```diff
@@ -11,14 +11,4 @@ class JobSerializer(serializers.ModelSerializer):
-        model = JobRun
-        fields = ('id', 'created_date', 'modified_date', 'config', 'status', 'result', 'type', 'log', 'error')
-
+        fields = ('id', 'created_date', 'modified_date', 'config', 'status', 'result', 'type', 'split', 'error')
```
You added `run`, don't you want it to be in the JSON?

TKasekamp commented on jobs/views.py:
```diff
@@ -120,18 +120,18 @@ def get_prediction(request, pk1, pk2):
                        }
-    log = Log.objects.get(pk=3)
-    jobrun=JobRun.objects.create(config=config, log=log, type=NEXT_ACTIVITY)
+    jobrun=Job.objects.create(config=config, run=True, type=NEXT_ACTIVITY)
```
This method looks like a hack. Update the description to say that this is a demo thing. Or add a TOD

TKasekamp commented on jobs/tasks.py:
```diff
@@ -28,14 +31,15 @@ def prediction(job,model):
+        
 def calculate(job,model):
     """ Main entry method for calculations"""
     #print("Start job {} with {}".format(job['type'], get_run(job)))
```
Did you comment out these print loggers? I'd rather like them to be working, same in `training`.

TKasekamp commented on jobs/migrations/0004_auto_20180326_1352.py:
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.0.3 on 2018-03-26 13:52
+
```
You can squash multiple migrations to one in PRs with https://docs.djangoproject.com/en/1.11/ref/django-admin/#django-admin-squashmigrations

Not that important, but makes the code cleaner.

TKasekamp commented on core/regression.py:
```diff
@@ -112,14 +111,3 @@ def prep_data(test_df):
     return test_data, original_test_data
-
-
-def __choose_regressor(regression_type: str):
```
Where is the regressor chosen if this is deleted?

TKasekamp commented on core/regression.py:
```diff
@@ -65,6 +47,23 @@ def kmeans_run(run_df, regressor, estimator):
+    return results
+
 def kmeans_clustering(original_test_data, regressor, estimator):
     print(estimator)
```
I don't like simple prints in the completed code. It makes it hard to see what is going on. Fine for local development, but remove them for PRs.

TKasekamp commented:
Sorry, I didn't notice you had made a PR. You should also remove unused imports, there should be a plugin for Eclipse for that.

TKasekamp commented:
A general suggestion: If you make a new method, you could add a Python doc comment under it to say that it was made by you. At the moment it's difficult for me to say whats mine and whats yours, especially for stuff in `core`.

stebranchi commented:
In these days I've already fixed some of these problems, for the choose regressor it's not needed in core because the model is created in training and core module simply takes the model from the database.
Now i'm trying to squash the migrations.
I discovered that i need to change the predmodel model because it's not good right now, so I think that it's better to wait to merge this PR. When i'll finish with predmodels I'll let you know.
For the comment in new method i'll do from now on, by the way I think I have changed like, everything.

TKasekamp commented:
I can't deploy the back-end to UT servers because the master version is broken. Please do a check of the end-to-end training tasks and fix it. 
```
11:54:02 TypeError: prediction() missing 1 required positional argument: 'model'
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/rq/worker.py", line 789, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 573, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 579, in _execute
    return self.func(*self.args, **self.kwargs)
TypeError: prediction() missing 1 required positional argument: 'model'
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/rq/worker.py", line 789, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 573, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.5/dist-packages/rq/job.py", line 579, in _execute
    return self.func(*self.args, **self.kwargs)
TypeError: prediction() missing 1 required positional argument: 'model'
```

stebranchi commented:
Try now if it works fine or not and let me know

TKasekamp commented:
I don't have the time at the moment to check out your repo and install everything. Frankly, you caused the bug and you have to fix it. 
Start the webserver and start the rqworker. Use the `curl` in the README to see if the task is working correctly. If you don't know how to use that then start the front-end and use the some random training options.

TKasekamp commented:
I have added you as collaborator to this and the front-end project. For development I'd like you to make a branch and then create a PR so I can see it. This is so we don't have a million repositories with the same code. And all the automated testing is on this repo.

Please don't directly commit stuff to the master branch 😄 

At the end, when you have to submit your thesis, you can make a fork.

stebranchi commented:
Wait a moment. I didn't ask you to solve the problem I caused , but simply have a feedback if I fixed it right or not, by the way I think I fixed it, because It works fine on my computer. Sorry if my previous message wasn't clear. 
At the moment i'm rewriting the code to work fine wih the new structure of PredModel

TKasekamp commented:
Ok sorry, if it works then I'd like to merge it so I could deploy it today. Or do you have any more refactoring you want to do here?

stebranchi commented:
I think that for now you can merge it.
